src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -345,13 +323,32 @@ protected virtual object CreateModel(ModelBindingContext bindingContext)
             // application developer should know that this was an invalid type to try to bind to.
             if (_modelCreator == null)
             {
+                // The following check causes the ComplexTypeModelBinder to NOT participate in binding structs as 
```

Comment:
I'd rather make `CanCreateModel()` `public virtual` than do this check so late. We're already `async` at this point and should have done all necessary checks in `CanCreateModel()`.

Could leave this method completely alone i.e. let it continue initializing `_modelCreator`. That way, someone who has overridden this method wouldn't have to change anything.